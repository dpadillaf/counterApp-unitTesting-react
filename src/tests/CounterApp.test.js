import { shallow } from "enzyme";
import CounterApp from "../CounterApp";


describe('test in CounterApp', () => {

    let wrapper = shallow( <CounterApp /> );

    beforeEach( () => {

        wrapper = shallow( <CounterApp /> );

    } );
    
    test('should mostrar CounterApp correctamnete', () => {
        
        const value = 20;
        const wrapper = shallow( <CounterApp value={ value } /> );

        expect( wrapper ).toMatchSnapshot();

    });

    test('should default value 100', () => {
        
        const defaultValue = Number(wrapper.find('h2').text());

        expect( defaultValue ).toBe( 100 );

    })
    
    test('should incrmenet value with btn +1', () => {
        
        wrapper.find( 'button' ).at(0).simulate('click');

        const counterText = wrapper.find('h2').text().trim();

        expect( counterText ).toBe( '101' );

    })
    
    test('should decrement value with btn -1', () => {
       
        wrapper.find( 'button' ).at(2).simulate('click');

        const counterText = wrapper.find('h2').text().trim();

        expect( counterText ).toBe( '99' );

    })

    test('should reset default value', () => {

        const wrapper = shallow( <CounterApp value={ 20 } /> );

        wrapper.find( 'button' ).at(0).simulate('click');
        wrapper.find( 'button' ).at(0).simulate('click');
        wrapper.find( 'button' ).at(1).simulate('click');

        const counterText = wrapper.find('h2').text().trim();

        expect( counterText ).toBe( '20' )

    })
    

})
