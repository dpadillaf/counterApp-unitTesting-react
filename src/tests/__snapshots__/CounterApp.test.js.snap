// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`test in CounterApp should mostrar CounterApp correctamnete 1`] = `
<Fragment>
  <h1>
    CounterApp
  </h1>
  <h2>
    20
  </h2>
  <button
    onClick={[Function]}
  >
    +1
  </button>
  <button
    onClick={[Function]}
  >
    Reset
  </button>
  <button
    onClick={[Function]}
  >
    -1
  </button>
</Fragment>
`;
